projects/plugins/jetpack/extensions/shared/external-media/index.js

```diff
@@ -43,8 +50,17 @@ if ( isCurrentUserConnected() && 'function' === typeof useBlockEditContext ) {
 		'external-media/replace-media-upload',
 		OriginalComponent => props => {
 			const { name } = useBlockEditContext();
+			const { registerInserterMediaCategory } = useDispatch( 'core/block-editor' );
 			let { render } = props;
 
+			useEffect( () => {
+				isGooglePhotosConnected( () =>
+					registerInserterMediaCategory?.( getGooglePhotosMediaCategory() )
+				);
+
+				registerInserterMediaCategory?.( getPexelsMediaCategory() );
+			}, [ registerInserterMediaCategory ] );
```

Comment:
Is the `editor.MediaUpload` hook the best place to register new categories in the media sidebar?

I was under the impression that `editor.MediaUpload` is used to replace the media upload form used by blocks such as the image or the video block, which apparently is unrelated to the media sidebar.

Reply:
Do you have any suggestions on where we could register the media categories?

Reply:
Does it work if we just register them at the of this file?

```diff
diff --git a/projects/plugins/jetpack/extensions/shared/external-media/index.js b/projects/plugins/jetpack/extensions/shared/external-media/index.js
index 51f92eef0a..7f7b097473 100644
--- a/projects/plugins/jetpack/extensions/shared/external-media/index.js
+++ b/projects/plugins/jetpack/extensions/shared/external-media/index.js
@@ -1,8 +1,7 @@
 import { isCurrentUserConnected } from '@automattic/jetpack-shared-extension-utils';
 import { useBlockEditContext } from '@wordpress/block-editor';
-import { useDispatch } from '@wordpress/data';
+import { dispatch } from '@wordpress/data';
 import { addFilter } from '@wordpress/hooks';
-import { useEffect } from 'react';
 import MediaButton from './media-button';
 import {
        getGooglePhotosMediaCategory,
@@ -50,17 +49,8 @@ if ( isCurrentUserConnected() && 'function' === typeof useBlockEditContext ) {
                'external-media/replace-media-upload',
                OriginalComponent => props => {
                        const { name } = useBlockEditContext();
-                       const { registerInserterMediaCategory } = useDispatch( 'core/block-editor' );
                        let { render } = props;
 
-                       useEffect( () => {
-                               isGooglePhotosConnected( () =>
-                                       registerInserterMediaCategory?.( getGooglePhotosMediaCategory() )
-                               );
-
-                               registerInserterMediaCategory?.( getPexelsMediaCategory() );
-                       }, [ registerInserterMediaCategory ] );
-
                        if ( isAllowedBlock( name, render ) || isFeaturedImage( props ) ) {
                                const { allowedTypes, gallery = false, value = [] } = props;
 
@@ -81,4 +71,10 @@ if ( isCurrentUserConnected() && 'function' === typeof useBlockEditContext ) {
                'external-media/individual-blocks',
                insertExternalMediaBlocks
        );
+
+       isGooglePhotosConnected( () =>
+               dispatch( 'core/block-editor' ).registerInserterMediaCategory?.( getGooglePhotosMediaCategory() )
+       );
+
+       dispatch( 'core/block-editor' ).registerInserterMediaCategory?.( getPexelsMediaCategory() );
 }

```